covidthailand.py

```diff
@@ -2776,7 +2868,7 @@ def save_plots(df):
         y=cols,
         kind="area",
         figsize=[20, 10],
-        colormap=AREA_COLOURS,
+        colormap=custom_cm(AREA_COLOURS, len(cols), last_colour='lightgrey'),
```

Comment:
could pass in unknown_colour into plot_area?

Reply:
We could pass the entire `colormap` into the plot_area function. Would that work?

```
def plot_area(df: pd.DataFrame, name: str, cols_subset: Union[str, List[str]], title: str, percent_fig: bool,
              unknown_name: str = 'Unknown', unknown_total: Union[None, str] = None, ma: bool = False, 
              cm: str = 'tab20', flip_cm: bool = False) -> None:


Creates one .png file and plots 2 charts, showing data in absolute numbers and percentage terms. 'cols_subset' argument can be used to select specific columns from the pandas DataFrame based on either a column name prefix or based on a list of column names.
```

What's missing is the `legend` argument, which contains all legend strings.

Reply:
Like this maybe?

```
def plot_area(df: pd.DataFrame, name: str, cols_subset: Union[str, List[str]], title: str, percent_fig: bool,
              legends: List[str], unknown_name: str = 'Unknown', unknown_total: Union[None, str] = None, 
              ma: bool = False, cm: str = 'tab20', flip_cm: bool = False) -> None:
```

Reply:
I like the idea of plot_area handling all the unknown stuff since its repeated over and over. I think we can assume plot_area always makes unknowns the same grey colour.

Reply:
I guess what I'm saying is there balance between adding some much manual specifiying arguments into helper function that its as long as the original code was :) it's tough to get that balance right.

Reply:
The best way I found so far is to use custom colormap (such as what is returned by the `custom_cm(...)` function. That should return a set of colours (including light grey if desired) that is consistent in all graphs. Let me work on it when converting the .png plots, one by one.

Reply:
@djay question is: do we keep the call to `custom_cm(...)` inside `plot_area` **or do we** move it outside and pass the Colormap object via an argument?

Reply:
I think keep it inside and if there is an unknown value being calculated then use it for this.